include/hmap/dynamic-hmap.hpp

```diff
@@ -337,6 +338,12 @@ class DynamicHMap {
 		}
 	}
 
+	[[noreturn]] void keyNotFound(const detail::KeyBase& k) const {
```

Comment:
I'm reasonably sure this can just be a `static` member instead of a `const` member.

Additionally, the implementation is not templated so please move it to the .cc file.